setup_ffe.sh

```diff
@@ -0,0 +1,244 @@
+
+rm -f my-libdatadog-build/arm64-darwin-24/include/datadog/crashtracker.h.bak*
+
+# Remove duplicates from ddsketch.h too
+echo "Removing duplicate types from ddsketch.h..."
+sed -i.bak1 '/typedef enum ddog_VoidResult_Tag {/,/} ddog_VoidResult;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/ddsketch.h
+sed -i.bak2 '/typedef struct ddog_VoidResult {/,/} ddog_VoidResult;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/ddsketch.h
+sed -i.bak3 '/typedef struct ddog_Vec_U8 {/,/} ddog_Vec_U8;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/ddsketch.h
+sed -i.bak4 '/typedef struct ddog_Error {/,/} ddog_Error;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/ddsketch.h
+rm -f my-libdatadog-build/arm64-darwin-24/include/datadog/ddsketch.h.bak*
+
+# Remove duplicates from datadog_ffe.h too
+echo "Removing duplicate types from datadog_ffe.h..."
+sed -i.bak1 '/typedef enum ddog_VoidResult_Tag {/,/} ddog_VoidResult;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/datadog_ffe.h
+sed -i.bak2 '/typedef struct ddog_VoidResult {/,/} ddog_VoidResult;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/datadog_ffe.h
+sed -i.bak3 '/typedef struct ddog_Vec_U8 {/,/} ddog_Vec_U8;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/datadog_ffe.h
+sed -i.bak4 '/typedef struct ddog_Error {/,/} ddog_Error;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/datadog_ffe.h
+rm -f my-libdatadog-build/arm64-darwin-24/include/datadog/datadog_ffe.h.bak*
+
+# Remove duplicates from library-config.h too
+echo "Removing duplicate types from library-config.h..."
+sed -i.bak1 '/typedef struct ddog_Vec_U8 {/,/} ddog_Vec_U8;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/library-config.h
+sed -i.bak2 '/typedef struct ddog_Error {/,/} ddog_Error;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/library-config.h
+sed -i.bak3 '/typedef struct ddog_Slice_CChar {/,/} ddog_Slice_CChar;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/library-config.h
+sed -i.bak4 '/typedef struct ddog_Slice_CChar ddog_CharSlice;/d' my-libdatadog-build/arm64-darwin-24/include/datadog/library-config.h
+rm -f my-libdatadog-build/arm64-darwin-24/include/datadog/library-config.h.bak*
+
+# Create minimal stub headers for libraries we're linking but don't actively use
+echo "Creating minimal stub headers for unused linked libraries..."
+
+# profiling.h - minimal stub  
+cat > my-libdatadog-build/arm64-darwin-24/include/datadog/profiling.h << 'EOF'
+#ifndef DDOG_PROFILING_H
+#define DDOG_PROFILING_H
+
+#pragma once
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include "common.h"
+
+// Minimal declarations for profiling functionality
+// Ruby bindings don't need full profiling API, just enough to link
+
+#ifdef __cplusplus
+extern "C" {
+#endif // __cplusplus
+
+// Stub function declarations - can be extended as needed
+
+#ifdef __cplusplus
+}  // extern "C"
+#endif  // __cplusplus
+
+#endif  /* DDOG_PROFILING_H */
+EOF
```

Comment:
Tip: You don't actually need to build profiling -- `DD_PROFILING_NO_EXTENSION=true` skips trying it to build at all.

Reply:
Incorporated this into the script though I'm not sure that I'm using it right.

Most of the profiling dependencies are in `ext/datadog_profiling_native_extension` which I'm understand that env variable lets us skip, but there's still one profiling dependency in `ext/libdatadog_api`

https://github.com/DataDog/dd-trace-rb/blob/f10f1df8e9e0d9a9e0683864897f9a57dfc088ed/ext/libdatadog_api/datadog_ruby_common.h#L8

So I still ended up needing to generate the `profiling.h` file

Reply:
Aaah oops that's actually an oversight from when `datadog_ruby_common.h` was extracted out of profiling: this can be replaced with `#include <datadog/common.h>`. I'll open a PR to fix this